core/lib/types/src/api/mod.rs

```diff
@@ -19,6 +19,14 @@ use crate::{
 
 pub mod en;
 
+/// Enum for choosing API mode
+#[derive(Copy, Clone, Debug, PartialEq, Display, Default)]
+pub enum APIMode {
+    #[default]
+    Modern,
+    Legacy,
```

Comment:
Bikeshedding: IMO, these variant names (and possibly, the enum name) are too vague. Do we plan to introduce other changes in the API in the future? If not (i.e., variability is confined to ETH transfer event handling), then at least variant naming should reflect that, e.g., `EthTransfersIncluded` / `EthTransfersExcluded`.

Reply:
I am not sure about it tbh, but from what @montekki said, I think yes: 

> For improved readability (which is which?) and future-proofing (agianst situations where we will be running more than two apis (we will).

Reply:
In this case, `Legacy` and `EthTransfersExcluded` would be better.